commons/src/main/java/com/github/globocom/viewport/commons/ViewPortPartialHelper.kt

```diff
@@ -0,0 +1,118 @@
+package com.github.globocom.viewport.commons
+
+import android.view.View
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.OrientationHelper
+import androidx.recyclerview.widget.RecyclerView
+import kotlin.math.abs
+
+object ViewPortPartialHelper {
+
+    fun findFirstPartiallyVisibleItemPosition(
+        recyclerView: RecyclerView,
+        linearLayoutManager: LinearLayoutManager,
+        threshold: Float = 1f
+    ): Int {
+        findOneVisibleChild(
+            linearLayoutManager,
+            0,
+            linearLayoutManager.childCount,
+            threshold
+        )?.let { childView ->
+            return recyclerView.getChildAdapterPosition(childView)
+        }
+        return RecyclerView.NO_POSITION
+    }
+
+    fun findLastPartiallyVisibleItemPosition(
+        recyclerView: RecyclerView,
+        linearLayoutManager: LinearLayoutManager,
+        threshold: Float = 1f
+    ): Int {
+        findOneVisibleChild(
+            linearLayoutManager,
+            linearLayoutManager.childCount - 1,
+            -1,
+            threshold
+        )?.let { childView ->
+            return recyclerView.getChildAdapterPosition(childView)
+        }
+        return RecyclerView.NO_POSITION
+    }
```

Comment:
E aqui poderia ficar assim 🙂
Substituindo os índices pelo `searchFromEndToStart` e utilizando o retorno direto (essa parte é só uma sugestão de estilo, então pode ignorar se quiser 😅, o importante é só os parâmetros mesmo)
```suggestion
    fun findFirstPartiallyVisibleItemPosition(
        recyclerView: RecyclerView,
        linearLayoutManager: LinearLayoutManager,
        threshold: Float = 1f
    ) = findOneVisibleChild(
            linearLayoutManager,
            threshold
        )?.let { childView ->
            recyclerView.getChildAdapterPosition(childView)
        } ?: RecyclerView.NO_POSITION

    fun findLastPartiallyVisibleItemPosition(
        recyclerView: RecyclerView,
        linearLayoutManager: LinearLayoutManager,
        threshold: Float = 1f
    ) = findOneVisibleChild(
            linearLayoutManager,
            threshold,
            true
        )?.let { childView ->
            recyclerView.getChildAdapterPosition(childView)
        } ?: RecyclerView.NO_POSITION
```